third-party/linshare/src/main/java/org/apache/james/linshare/client/LinshareAPI.java

```diff
@@ -0,0 +1,115 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.linshare.client;
+
+import static org.apache.james.linshare.client.LinshareAPI.Header.ACCEPT_APPLICATION_JSON;
+import static org.apache.james.linshare.client.LinshareAPI.Header.CONTENT_TYPE_APPLICATION_JSON;
+import static org.apache.james.linshare.client.LinshareAPI.Header.CONTENT_TYPE_MULTIPART;
+
+import java.io.File;
+import java.util.List;
+
+import org.apache.james.linshare.LinshareConfiguration;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+import feign.Feign;
+import feign.Headers;
+import feign.Logger;
+import feign.Param;
+import feign.RequestInterceptor;
+import feign.RequestLine;
+import feign.RequestTemplate;
+import feign.form.FormEncoder;
+import feign.jackson.JacksonDecoder;
+import feign.jackson.JacksonEncoder;
+import feign.slf4j.Slf4jLogger;
+
+public interface LinshareAPI {
+
+    interface Header {
+        String ACCEPT_APPLICATION_JSON = "Accept: application/json";
+        String CONTENT_TYPE_MULTIPART = "Content-Type: multipart/form-data";
+        String CONTENT_TYPE_APPLICATION_JSON = "Content-Type: application/json";
+    }
+
+    class AuthorizationInterceptor implements RequestInterceptor {
+
+        private final String authorizationToken;
```

Comment:
Should be of type AuthorizationToken (let's keep the strong types to the edge of our code)